src/braket/circuits/instruction.py

```diff
@@ -81,6 +83,24 @@ def target(self) -> QubitSet:
         """
         return self._target
 
+    def adjoint(self) -> Instruction:
+        """Returns an Instruction with the adjoint of this instruction's own operator
+
+        This operation only works on Gate operators.
+
+        Returns:
+            Instruction: A new instruction whose operator is the adjoint of `operator`
+
+        Raises:
+            NotImplementedError: If `operator` is not of type `Gate` or `CompilerDirective`
+        """
+        operator = self._operator
+        if isinstance(operator, Gate):
+            return Instruction(operator.adjoint(), self._target)
+        elif isinstance(operator, CompilerDirective):
```

Comment:
I'm not sure that this makes sense generally.

Reply:
How so? Once if we start having more types, I suppose we can do a `singledispatch` like we do for translators.

Reply:
It just seems like not all CompilerDirectives will have counterparts. But (I think) I get your point that we can introduce a new class that distinguishes, if we need to.

Reply:
Yep. In a bit more detail, all `CompilerDirective`s need to be in some `Instruction` (and compilation-relate construct that _doesn't_ go in an `Instruction` would need a new class), and `counterpart` makes sense generally for anything in an instruction; it doesn't need to be symmetric, with an open to a close like with verbatim boxes, but anything that can be reversed in some way.